packages/bpk-docs/src/components/neo/DocsPageWrapper/DocsPageWrapper.js

```diff
@@ -0,0 +1,126 @@
+import BpkSmallMobileIcon from 'bpk-component-icon/sm/mobile';
+import BpkSmallWindowIcon from 'bpk-component-icon/sm/window';
+import { withButtonAlignment } from 'bpk-component-icon';
+import { cssModules } from 'bpk-react-utils';
+import isString from 'lodash/isString';
+
+import Heading from './../Heading';
+
+import STYLES from './DocsPageWrapper.scss';
+import Paragraph from '../Paragraph';
+
+const getClassName = cssModules(STYLES);
+
+const AlignedBpkSmallMobileIcon = withButtonAlignment(BpkSmallMobileIcon);
+const AlignedBpkSmallWindowIcon = withButtonAlignment(BpkSmallWindowIcon);
+
+const childrenPropType = PropTypes.oneOfType([
+  PropTypes.arrayOf(PropTypes.node),
+  PropTypes.node,
+]);
+
+const contentShape = PropTypes.oneOfType([PropTypes.string, childrenPropType]);
+
+const createBlurb = content =>
+  isString(content) ? <Paragraph>{content}</Paragraph> : content;
+
+class DocsPageWrapper extends React.Component {
+  static propTypes = {
+    title: PropTypes.string.isRequired,
+    blurb: contentShape,
+    webSubpage: PropTypes.element,
+    nativeSubpage: PropTypes.element,
+  };
+
+  static defaultProps = {
+    blurb: null,
+    webSubpage: null,
+    nativeSubpage: null,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { selected: 'native' };
+  }
+
+  render() {
+    const { blurb, nativeSubpage, title, webSubpage } = this.props;
+
+    const showPlatformSwitcher = webSubpage && nativeSubpage;
+    const selectedSubpage = webSubpage || nativeSubpage;
+
+    return (
+      <BpkContentContainer className={getClassName('bpkdocs-page-wrapper')}>
+        <div className={getClassName('bpkdocs-page-wrapper__inner')}>
+          <Heading level="h1">{title}</Heading>
+          {blurb && createBlurb(blurb)}
+        </div>
+        {showPlatformSwitcher && (
+          <div>
+            <BpkHorizontalNav
+              className={getClassName(
+                'bpkdocs-page-wrapper__platform-switcher',
+              )}
+            >
+              <BpkHorizontalNavItem
+                name="native"
+                selected={this.state.selected === 'native'}
+                onClick={() => this.setState({ selected: 'native' })}
+              >
+                <AlignedBpkSmallMobileIcon
+                  className={getClassName(
+                    'bpkdocs-page-wrapper__platform-icon',
+                  )}
+                />
+                Native
+              </BpkHorizontalNavItem>
+              <BpkHorizontalNavItem
+                name="web"
+                selected={this.state.selected === 'web'}
+                onClick={() => this.setState({ selected: 'web' })}
+              >
+                <AlignedBpkSmallWindowIcon
+                  className={getClassName(
+                    'bpkdocs-page-wrapper__platform-icon',
+                  )}
+                />
+                Web
+              </BpkHorizontalNavItem>
+            </BpkHorizontalNav>
+            <div>
```

Comment:
does this wrapper div need to be there?

Reply:
Hmm, let's find out!

Reply:
It does, because of React's hatred of adjoining elements not wrapped in an enclosing element:

```
<BpkHorizontalNav>
    // Nav stuff
</BpkHorizontalNav>
<div>
  // We're rendering the selected web/native readme in here.
</div>
```